src/beehive-stack.ts

```diff
@@ -1,12 +1,80 @@
 import * as cdk from '@aws-cdk/core'
+import { CloudFrontAllowedMethods, CloudFrontWebDistribution, OriginAccessIdentity, ViewerCertificate } from '@aws-cdk/aws-cloudfront'
+import { CnameRecord, HostedZone } from '@aws-cdk/aws-route53'
+import { Bucket } from '@aws-cdk/aws-s3'
 import { SharedServiceStackProps } from './shared-stack-props'
+import { FoundationStack } from './foundation-stack'
 
-export interface BeehiveStackProps extends SharedServiceStackProps {}
+export interface BeehiveStackProps extends SharedServiceStackProps {
+  readonly contextEnvName: string,
+  readonly domainStackName: string,
+  readonly hostnamePrefix: string,
+  readonly createDns: boolean,
+  readonly foundationStack: FoundationStack
+}
 
 export class BeehiveStack extends cdk.Stack {
+  public readonly hostname: string
+  public readonly bucket: Bucket
+  public readonly cloudfront: CloudFrontWebDistribution
+
   constructor (scope: cdk.Construct, id: string, props: BeehiveStackProps) {
     super(scope, id, props)
 
     // The code that defines your stack goes here
+
+    this.hostname = `${props.hostnamePrefix || this.stackName}`
+    const webBucket = new Bucket(this, 'beehiveBucket', {
+      serverAccessLogsBucket: props.foundationStack.logBucket,
+      serverAccessLogsPrefix: `s3/${this.hostname}`,
+    })
+
+    this.cloudfront = new CloudFrontWebDistribution(this, 'beehiveDistribution', {
+      comment: this.hostname,
+      aliasConfiguration: {
+        names: [props.hostnamePrefix + '.' + cdk.Fn.importValue(`${props.domainStackName}:DomainName`)],
```

Comment:
This has the opportunity to be different from the exposed this.hostname

Reply:
It does, but is that an issue? If it is, It should probably be an update to the definition of `this.hostname` rather than a change at this alias, which is taking a prefix and then appending the actual domain to the end to create (hopefully) a known-good FQDN. Thoughts?

Reply:
Seems like you'd want this instead:
```suggestion
        names: [this.hostname + '.' + cdk.Fn.importValue(`${props.domainStackName}:DomainName`)],
```

Reply:
Updated